src/pcm/SoxrResampler.cxx

```diff
@@ -109,14 +218,32 @@ SoxrPcmResampler::Open(AudioFormat &af, unsigned new_sample_rate)
 	assert(audio_valid_sample_rate(new_sample_rate));
 
 	soxr_error_t e;
+	soxr_io_spec_t* p_soxr_io = nullptr;
+	if(soxr_io_custom_recipe.scale != 0) {
```

Comment:
This is still not an explicit check for "custom recipe".
You initialized the global variable, but it is not guaranteed per soxr API documentation. Looking and the soxr source code, this looks completely wrong, because `soxr_io_spec()` initializes `scale=1`. Did you ever test this?
And: comparing floating-point values with `==` is conceptually broken. You should avoid this.

Reply:
Yes you are right about the scale=1 vs 0 mixing up linear vs db.
Will add static variable soxr_use_custom_recipe for explit use.